src/lib.rs

```diff
@@ -305,19 +313,43 @@ pub fn run_with(cmd_args: Option<&[&str]>) -> Result<(), String> {
         );
     }
 
+    let paths = fs::read_dir(&args.nodes_dir)
+        .map_err(|_| "Could not read existing testnet log dir".to_string())?;
+
+    let existing_nodes_count = paths
+        .collect::<Result<Vec<_>, io::Error>>()
+        .map_err(|_| "Error collecting testnet log dir".to_string())?
+        .len();
+
+    println!("{:?} existing nodes found", existing_nodes_count);
+
+    // either we have genesis only, or all existing ndoes
+    let start = existing_nodes_count;
```

Comment:
Would it be better to bail `if existing_nodes_count == 0 && adding_nodes` ? Meaning there's no genesis to add nodes to.

Reply:
bail sure. i wouldn't try again, if the first failed, there's something bigger afoot